### PR TITLE
[bridge indexer] - record bridge txn errors

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

Describe the changes or additions included in this PR.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-bridge-indexer/src/sui_worker.rs [hunk 1]
```diff
@@ -65,28 +66,41 @@ impl SuiBridgeWorker {
...
+                    Ok(vec![ProcessedTxnData::Error(SuiTxnError {
+                        tx_digest: *tx.transaction.digest(),
+                        sender: tx.transaction.sender_address(),
+                        timestamp_ms,
+                        failure_status: error.to_string(),
+                        cmd_idx: command.map(|idx| idx as u64),
+                    })])
+                } else {
+                    Ok(vec![])
+                }
```
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/sui_worker.rs:102]: are we interested in txes that succeed but do not have events?

File: crates/sui-bridge-indexer/src/sui_transaction_handler.rs [hunk 2]
```diff
@@ -34,7 +36,8 @@ pub async fn handle_sui_transactions_loop(
     let mut stream = ReceiverStream::new(rx).ready_chunks(checkpoint_commit_batch_size);
     while let Some(batch) = stream.next().await {
         // unwrap: batch must not be empty
-        let cursor = batch.last().unwrap().1;
+        let (txns, cursor) = batch.last().cloned().unwrap();
+        let last_ckp = txns.last().map(|tx| tx.checkpoint).unwrap_or_default();
```
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/sui_transaction_handler.rs:40]: why is this line moved here?

Comment (patrickkuo) [reply crates/sui-bridge-indexer/src/sui_transaction_handler.rs]: this was probably needed for some other reason previously, will revert it.

File: crates/sui-bridge-indexer/src/sui_transaction_handler.rs [hunk 3]
```diff
@@ -43,8 +45,8 @@ pub async fn handle_sui_transactions_loop(
 
         // write batched token transfers to DB
         if !token_transfers.is_empty() {
+            let last_ckp = txns.last().map(|tx| tx.checkpoint).unwrap_or_default();
```
Comment (longbowlu) [inline crates/sui-bridge-indexer/src/sui_transaction_handler.rs:48]: this probably works too, but 
```
let last_ckp = token_transfers.last().as_ref().unwrap().block_height;
```
feel more straightforward mentally

**Review by longbowlu — APPROVED** [review]
please take a look at https://github.com/MystenLabs/sui/pull/18510/files#r1678621362
